qiskit_nature/second_q/operators/spin_op.py

```diff
@@ -18,532 +18,602 @@
+            idx = int(index_split[0])
+            exp = int(index_split[1]) if len(index_split) > 1 else 1
+            yield char, idx, exp
+
+    @classmethod
+    def x(cls, spin: float | Fraction = Fraction(1, 2)) -> SpinOp:
+        """Constructs the X spin operator for a given spin.
 
         Returns:
-            Spin number
+            The X spin operator for `spin`.
         """
-        return Fraction(self._dim - 1, 2)
+        return cls({"X_0": 1.0}, spin=spin, copy=False)
 
-    @property
-    def x(self) -> np.ndarray:
-        """A np.ndarray storing the power i of (spin) X operators on the spin system.
-        I.e. [0, 4, 2] corresponds to X_0^0 \\otimes X_1^4 \\otimes X_2^2, where X_i acts on the
-        i-th spin system in the register.
+    @classmethod
+    def y(cls, spin: float | Fraction = Fraction(1, 2)) -> SpinOp:
+        """Constructs the Y spin operator for a given spin.
+
+        Returns:
+            The Y spin operator for `spin`.
         """
-        return self._spin_array[0]
+        return cls({"Y_0": 1.0}, spin=spin, copy=False)
 
-    @property
-    def y(self) -> np.ndarray:
-        """A np.ndarray storing the power i of (spin) Y operators on the spin system.
-        I.e. [0, 4, 2] corresponds to Y_0^0 \\otimes Y_1^4 \\otimes Y_2^2, where Y_i acts on the
-        i-th spin system in the register.
+    @classmethod
+    def z(cls, spin: float | Fraction = Fraction(1, 2)) -> SpinOp:
+        """Constructs the Z spin operator for a given spin.
+
+        Returns:
+            The Z spin operator for `spin`.
         """
-        return self._spin_array[1]
+        return cls({"Z_0": 1.0}, spin=spin, copy=False)
 
-    @property
-    def z(self) -> np.ndarray:
-        """A np.ndarray storing the power i of (spin) Z operators on the spin system.
-        I.e. [0, 4, 2] corresponds to Z_0^0 \\otimes Z_1^4 \\otimes Z_2^2, where Z_i acts on the
-        i-th spin system in the register.
+    @classmethod
+    def one(cls, spin: float | Fraction = Fraction(1, 2)) -> SpinOp:
+        # pylint: disable=arguments-differ
+        """Constructs the "one" spin operator for a given spin.
+
+        Returns:
+            The "one" spin operator for `spin`.
         """
-        return self._spin_array[2]
+        return cls({"": 1.0}, spin=spin, copy=False)
 
-    def add(self, other: "SpinOp") -> "SpinOp":
-        if not isinstance(other, SpinOp):
-            raise TypeError(
-                "Unsupported operand type(s) for +: 'SpinOp' and " f"'{type(other).__name__}'"
-            )
+    @classmethod
+    def zero(cls, spin: float | Fraction = Fraction(1, 2)) -> SpinOp:
+        # pylint: disable=arguments-differ
+        """Constructs the "zero" spin operator for a given spin.
 
-        if self.register_length != other.register_length:
-            raise TypeError("Incompatible register lengths for '+'.")
+        Returns:
+            The "one" spin operator for `spin`.
```

Comment:
```suggestion
            The "zero" spin operator for `spin`.
```
Copy/paste just not updated I imagine